gneiss/sort.py

```diff
@@ -114,3 +114,118 @@ def niche_sort(table, gradient, niche_estimator=mean_niche_estimator):
     table = table.reindex(index=gradient.index,
                           columns=est_niche.index)
     return table
+
+
+def ladderize(tree, ascending=True):
+    """
+    Sorts tree according to the size of the subtrees.
+
+    Parameters
+    ----------
+    tree : skbio.TreeNode
+       Input tree where leafs correspond to features.
+
+    Returns
+    -------
+    skbio.TreeNode
+       A tree whose tips are sorted according to subtree size.
+
+    Examples
+    --------
+    >>> from skbio import TreeNode
+    >>> tree = TreeNode.read([u'((a,b)c, ((g,h)e,f)d)r;'])
+    >>> print(tree.ascii_art())
+                        /-a
+              /c-------|
+             |          \-b
+    -r-------|
+             |                    /-g
+             |          /e-------|
+              \d-------|          \-h
+                       |
+                        \-f
+    >>> sorted_tree = ladderize(tree)
+    >>> print(sorted_tree.ascii_art())
+                        /-a
+              /c-------|
+             |          \-b
+    -r-------|
+             |          /-f
+              \d-------|
+                       |          /-g
+                        \e-------|
+                                  \-h
+    """
+    sorted_tree = tree.copy()
+    # Note that this is operation is not optimal
+    # See https://github.com/biocore/gneiss/issues/58
+    for n in sorted_tree.postorder(include_self=True):
+        sizes = [len(list(k.tips())) for k in n.children]
```

Comment:
This isn't critical for this PR but this results in a full traversal of a clade for every single node. It would be better to cache ntips across the tree with a single traversal, eg:

```python
def cache_ntips(tree):
    for n in tree.postorder(include_self=True):
        if n.is_tip():
            n_tips = 1
        else:
           n_tips = sum(c.n_tips for c in n.children)
```